core/bindings.rs

```diff
@@ -162,17 +163,43 @@ pub fn initialize_context<'s>(
 
   let scope = &mut v8::ContextScope::new(scope, context);
 
-  // global.Deno = { core: {} };
   let deno_key = v8::String::new(scope, "Deno").unwrap();
+  let core_key = v8::String::new(scope, "core").unwrap();
+  let ops_key = v8::String::new(scope, "ops").unwrap();
+  // Snapshot already registered `Deno.core.ops` but
+  // extensions may provide ops that aren't part of the snapshot.
```

Comment:
This is a red flag for me. What if you want to create some ops during snapshotting but don't want for them to available after the snapshot is loaded? AFAIK this can happen in TSC snapshot which needs some ops to create a snapshot, but they should be exposed during runtime. 

Please explain this bit

Reply:
> What if you want to create some ops during snapshotting but don't want for them to be available after the snapshot is loaded?
 
Sure, no problem with that. It won't be available because the function reference expires.

> AFAIK this can happen in TSC snapshot which needs some ops to create a snapshot, but they should be exposed during runtime.

Yeah, this feature is added to support TSC ops. This handles the case when ops are not part of the snapshot but you still need them at runtime (during build.rs)